api/specs.proto

```diff
@@ -364,3 +368,13 @@ message SecretSpec {
 	// Data is the secret payload - the maximum size is 500KB (that is, 500*1024 bytes)
 	bytes data = 2;
 }
+
+// ConfigSpec specifies user-provided configuration files.
+message ConfigSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	// Data is the config payload - the maximum size is 500KB (that is, 500*1024 bytes)
```

Comment:
Agreed with @dhiltgen on both counts. Single file, and opt-in templating.

I feel strongly about the templating though. If we don't ship it, we will end up w/ secrets and configs being treated exactly the same.